http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcRequestAuthenticator.java

```diff
@@ -180,15 +216,47 @@ protected String getRedirectUri(String state) {
             if (deployment.getAuthUrl() == null) {
                 return null;
             }
-            URIBuilder redirectUriBuilder = new URIBuilder(deployment.getAuthUrl())
-                    .addParameter(RESPONSE_TYPE, CODE)
-                    .addParameter(CLIENT_ID, deployment.getResourceName())
-                    .addParameter(REDIRECT_URI, rewrittenRedirectUri(url))
-                    .addParameter(STATE, state);
-            redirectUriBuilder.addParameters(forwardedQueryParams);
+
+            String redirectUri = rewrittenRedirectUri(url);
+            URIBuilder redirectUriBuilder = new URIBuilder(deployment.getAuthUrl());
+            redirectUriBuilder.addParameter(RESPONSE_TYPE, CODE)
+                    .addParameter(CLIENT_ID, deployment.getResourceName());
+            if (deployment.getAuthenticationRequestFormat().contains(REQUEST_TYPE_OAUTH2)) {
+                redirectUriBuilder.addParameter(REDIRECT_URI, redirectUri)
+                        .addParameter(STATE, state)
+                        .addParameters(forwardedQueryParams);
+            } else if (deployment.getAuthenticationRequestFormat().contains(REQUEST_TYPE_REQUEST_URI)) {
+                if (deployment.getRequestUriParameterSupported()) {
+                    String request = convertToRequestParameter(redirectUriBuilder, redirectUri, state, forwardedQueryParams);
+                    String request_uri = getRequestUri(request);
+                    redirectUriBuilder.addParameter(REQUEST_URI, request_uri);
+                    redirectUriBuilder.addParameter(REDIRECT_URI, redirectUri);
+                } else {
+                    // send request as usual
+                    redirectUriBuilder.addParameter(REDIRECT_URI, redirectUri)
+                            .addParameter(STATE, state)
+                            .addParameters(forwardedQueryParams);
+                    log.info("The OpenID provider does not support the request parameter. Sending the request using OAuth2 format.");
+                }
+            } else if (deployment.getAuthenticationRequestFormat().contains(REQUEST_TYPE_REQUEST)) {
+                if (deployment.getRequestParameterSupported()) {
+                    // add request objects into request parameter
+                    String request = convertToRequestParameter(redirectUriBuilder, redirectUri, state, forwardedQueryParams);
+                    redirectUriBuilder.addParameter(REDIRECT_URI, redirectUri);
+                    redirectUriBuilder.addParameter(REQUEST, request);
+                } else {
+                    // send request as usual
+                    redirectUriBuilder.addParameter(REDIRECT_URI, redirectUri)
+                            .addParameter(STATE, state)
+                            .addParameters(forwardedQueryParams);
+                    log.info("The OpenID provider does not support the request_uri parameter. Sending the request using OAuth2 format.");
```

Comment:
This should make use of ElytronMessages.

Reply:
fixed